src/tocsym.c

```diff
@@ -285,7 +290,10 @@ Symbol *toSymbol(Dsymbol *s)
                 Symbol *s = symbol_calloc(id);
                 slist_add(s);
 
-                s->prettyIdent = fd->toPrettyChars(true);
+                if (global.params.isWindows)
+                    s->prettyIdent = fd->toPrettyChars(true);
+                else
+                    s->prettyIdent = fd->toChars();
                 s->Sclass = SCglobal;
```

Comment:
Should just be the base identifier name, I've never seen strange things coming from gdc when using this, can check instantiated functions just to be sure though.  Why do you think it would require half of a D parser to do it this way though?
If dmd is emitting children correctly in debug, gdb can build up a list of qualified names just by looking up the names of all parents, joining them together using the appropriate separator.